src/components/Scorecards/ScorecardDetailsPage/ScorecardsTableCard/ScorecardsTableCard.tsx

```diff
@@ -13,43 +13,106 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import React from 'react';
+import React, { useMemo } from 'react';
+import { EmptyState, InfoCard, Table as BSTable, TableColumn } from '@backstage/core-components';
 import { ScorecardServiceScore } from '../../../../api/types';
 import { useDetailCardStyles } from '../../../../styles/styles';
-import { Table, TableBody } from '@material-ui/core';
-import { ScorecardsTableRow } from './ScorecardsTableRow';
-import { EmptyState, InfoCard } from '@backstage/core-components';
+import { Gauge } from '../../../Gauge';
+import { humanizeAnyEntityRef } from '../../../../utils/types';
+import { defaultComponentRefContext } from '../../../../utils/ComponentUtils';
+import { ScorecardLadderLevelBadge } from '../../../Common/ScorecardLadderLevelBadge';
+import { ScorecardServiceRefLink } from '../../../ScorecardServiceRefLink/ScorecardServiceRefLink';
 
 interface ScorecardsTableProps {
   scorecardId: number;
   scores: ScorecardServiceScore[];
 }
 
+const PAGE_SIZE = 25;
+
+const columns: TableColumn[] = [{
+  field: 'scorePercentage',
+  sorting: true,
+  title: 'Score',
+  render: (data: { scorePercentage?: number }) => {
+    return (
+      <Gauge
+        value={data?.scorePercentage ?? 0}
+        strokeWidth={10}
+        trailWidth={10}
+      />
+    )
+  }
+}, {
+  customFilterAndSearch: (filter, rowData: { serviceName?: string }) => {
+    return rowData.serviceName?.indexOf(filter) !== -1;
+  },
+  customSort: ((data1: { serviceName?: string }, data2: { serviceName?: string }) => {
+    return (data1?.serviceName ?? '')?.localeCompare(data2?.serviceName ?? '')
+  }),
+  field: 'name',
+  highlight: true,
+  sorting: true,
+  title: 'Service name',
+}, {
+  field: 'level',
+  sorting: false,
+  title: 'Level',
+}];
+
 export const ScorecardsTableCard = ({
   scorecardId,
   scores,
 }: ScorecardsTableProps) => {
   const classes = useDetailCardStyles();
 
+  const data = useMemo(() => {
+    return scores.map((score) => {
+      const currentLevel = score.ladderLevels?.[0]?.currentLevel;
+      const serviceName = humanizeAnyEntityRef(
+        score.componentRef,
+        defaultComponentRefContext,
+      );
+      return {
+        level: currentLevel ? (
+          <ScorecardLadderLevelBadge
+            name={currentLevel.name}
+            color={currentLevel.color}
+          />) : null,
+        name: (
+          <ScorecardServiceRefLink
+            scorecardId={scorecardId}
+            componentRef={score.componentRef}
+          >
+            {serviceName}
+          </ScorecardServiceRefLink>
+        ),
+        scorePercentage: score.scorePercentage,
+        serviceName, // for filtering
```

Comment:
just as a nitpick, i'd opt for having the data just be objects, and have custom renders in each of the columns, like i changed for `scorePercentage`, instead of having the non-rendered additional data -- but pushing through for time sake 😄

Reply:
Good idea 👍. I didn't think of that.